pkg/overlay/cache.go

```diff
@@ -267,6 +256,33 @@ func (cache *Cache) GetAll(ctx context.Context, ids storj.NodeIDList) (_ []*Node
 	return cache.db.GetAll(ctx, ids)
 }
 
+// UnreliableOrOffline filters a set of nodes to unhealth or offlines node, independent of new
+func (cache *Cache) UnreliableOrOffline(ctx context.Context, nodeIds storj.NodeIDList) (badNodes map[storj.NodeID]bool, err error) {
+	defer mon.Task()(&ctx)(&err)
+	badNodes = make(map[storj.NodeID]bool)
+	goodNodes, err := cache.ReliableAndOnline(ctx, nodeIds)
+	if err == nil {
```

Comment:
I feel that's easy to understand the logic
```
if err != nil {
	return nil, err
}
...
..

return badNodes, nil
```

because the logic inside of the conditional is shorter and very concise and clarifies that the error is returned at that point and otherwise there is not returned error, but not just because it's consistent with the rest of the code base.

It's a personal opinion and the existent comment motivated me to comment too.

Reply:
Fixed